colossus/src/main/scala/colossus/protocols/http/HttpRequestParser.scala

```diff
@@ -16,11 +16,13 @@ object HttpRequestParser {
   def apply(size: DataSize = DefaultMaxSize) = maxSize(size, httpRequest)
 
   protected def httpRequest: Parser[HttpRequest] = httpHead |> {head => 
-    if (head.contentLength > 0) {
-      bytes(head.contentLength) >> {body => HttpRequest(head, Some(body))}
-    } else {
-      const(HttpRequest(head, None))
-    }
+    head.singleHeader(HttpHeaders.TransferEncoding) match { 
+      case None | Some("identity") => head.contentLength match {
```

Comment:
Not that I have a strong feeling about it, but do we want to handle a case where both content length and transfer encoding are present in the header, and possibly emit a failure?